src/core/net/ip6.cpp

```diff
@@ -536,20 +536,17 @@ otError Ip6::SendDatagram(Message &aMessage, MessageInfo &aMessageInfo, uint8_t
         SuccessOrExit(error = AddTunneledMplOption(aMessage, header, aMessageInfo));
     }
 
```

Comment:
Adding a note on the `diff` below:
- Instead of after `exit` checking `if (error == OT_ERROR_NONE)` and then doing the queuing/fragmentation, we do it before the `exit` label.
- In the code above, all jumps to `exit` indicate a failure and would set the `error`.